x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/tree/Tree.java

```diff
@@ -257,6 +257,12 @@ public void validate() {
         detectCycle();
     }
 
+    @Override
+    public long estimatedNumOperations() {
+        // Grabbing the features from the doc + the depth of the tree
+        return (long)Math.ceil(Math.log(nodes.size())) + featureNames.size();
```

Comment:
It's probably worth making `nodes.isEmpty() == false` an invariant for this class.  This line is assuming that `nodes.size() > 0`, as is line 168.  But there are a couple of other places in the file where there are `if (nodes.isEmpty()) {` checks.  Since it's clearly never intended to be empty it's probably best to validate that in the constructor and not bother to check anywhere else.

Reply:
@droberts195 sounds good to me :D.